test/ssh/sshServer.test.js

```diff
@@ -0,0 +1,341 @@
+        };
+
+        mockDb.findUser.resolves({
+          username: 'test-user',
+          password: '$2a$10$mockHash',
+        });
+
+        const bcrypt = require('bcryptjs');
+        sinon.stub(bcrypt, 'compare').resolves(true);
+
+        server.handleClient(mockClient);
+        const authHandler = mockClient.on.withArgs('authentication').firstCall.args[1];
+        await authHandler(mockCtx);
+
+        expect(mockDb.findUser.calledWith('test-user')).to.be.true;
+        expect(bcrypt.compare.calledWith('test-password', '$2a$10$mockHash')).to.be.true;
+        expect(mockCtx.accept.calledOnce).to.be.true;
+      });
+    });
+  });
+
+  describe('handleSession', () => {
+    let mockSession;
+    let mockStream;
+    let mockAccept;
+    let mockReject;
+
+    beforeEach(() => {
+      mockStream = {
+        write: sinon.stub(),
+        end: sinon.stub(),
+        exit: sinon.stub(),
+        on: sinon.stub(),
+      };
+
+      mockSession = {
+        on: sinon.stub(),
+        _channel: {
+          _client: {
+            userPrivateKey: null,
+          },
+        },
+      };
+
+      mockAccept = sinon.stub().returns(mockSession);
+      mockReject = sinon.stub();
+    });
+
+    it('should handle git-upload-pack command', async () => {
+      const mockInfo = {
+        command: "git-upload-pack 'test/repo'",
+      };
+
+      mockChain.executeChain.resolves({
+        error: false,
+        blocked: false,
+      });
+
+      const { Client } = require('ssh2');
+      const mockSsh2Client = {
+        on: sinon.stub(),
+        connect: sinon.stub(),
+        exec: sinon.stub(),
+      };
+
+      // Mock the SSH client constructor
+      sinon.stub(Client.prototype, 'on').callsFake(mockSsh2Client.on);
+      sinon.stub(Client.prototype, 'connect').callsFake(mockSsh2Client.connect);
+      sinon.stub(Client.prototype, 'exec').callsFake(mockSsh2Client.exec);
+
+      // Mock the ready event
+      mockSsh2Client.on.withArgs('ready').callsFake((event, callback) => {
+        callback();
+      });
+
+      // Mock the exec response
+      mockSsh2Client.exec.callsFake((command, options, callback) => {
+        const mockStream = {
+          on: sinon.stub(),
+          write: sinon.stub(),
+          end: sinon.stub(),
+        };
+        callback(null, mockStream);
+      });
+
+      server.handleSession(mockAccept, mockReject);
+      const execHandler = mockSession.on.withArgs('exec').firstCall.args[1];
+      await execHandler(mockAccept, mockReject, mockInfo);
+
+      expect(
+        mockChain.executeChain.calledWith({
+          method: 'GET',
+          originalUrl: " 'test/repo",
+          isSSH: true,
+          headers: {
+            'user-agent': 'git/2.0.0',
+            'content-type': undefined,
+          },
+        }),
+      ).to.be.true;
+    });
+
+    it('should handle git-receive-pack command', async () => {
+      const mockInfo = {
+        command: "git-receive-pack 'test/repo'",
```

Comment:
Other possible test cases for `handleSession`:

- Invalid repo handling
  - Currently gets stuck on `[GitHub SSH Debug] Client: Trying github.com on port 22 ...` without throwing error. `executeChain` doesn't seem to get called
- Edge case tests:
  - Empty repository name, whitespace inputs
  - SSH injection (example: `command: "git-upload-pack 'repo'; rm -rf /; #",`)